pkg/util/common.go

```diff
@@ -1309,58 +1366,151 @@ func IsPodReady(clientset *kubernetes.Clientset, namespace string, labelSelector
 	return true, nil
 }
 
-// GetOperatorNamespace returns the namespace of Synopsys Operator based on the labels
-func GetOperatorNamespace(clientset *kubernetes.Clientset) (string, error) {
+// GetClusterScope returns whether any of the CRD is cluster scope
+func GetClusterScope(apiExtensionClient *apiextensionsclient.Clientset) bool {
+	crds := []string{AlertCRDName, BlackDuckCRDName, OpsSightCRDName, PrmCRDName}
+	for _, crd := range crds {
+		cr, err := GetCustomResourceDefinition(apiExtensionClient, crd)
+		if err == nil && strings.EqualFold("CLUSTER", string(cr.Spec.Scope)) {
+			return true
+		}
+	}
+	return false
+}
+
+// GetOperatorNamespace returns the namespace of the synopsys operator based on the labels
+func GetOperatorNamespace(clientset *kubernetes.Clientset, namespace string) (string, error) {
 	// check if operator is already installed
-	rcs, err := ListReplicationControllers(clientset, metav1.NamespaceAll, "app=synopsys-operator,component=operator")
+	rcs, err := ListReplicationControllers(clientset, namespace, "app=synopsys-operator,component=operator")
+	if err == nil && len(rcs.Items) > 0 {
+		for _, rc := range rcs.Items {
+			return rc.Namespace, nil
+		}
+	}
+	deployments, err := ListDeployments(clientset, namespace, "app=synopsys-operator,component=operator")
+	if err == nil && len(deployments.Items) > 0 {
+		for _, deployment := range deployments.Items {
+			return deployment.Namespace, nil
+		}
+	}
+	pods, err := ListPodsWithLabels(clientset, namespace, "app=synopsys-operator,component=operator")
+	if err == nil && len(pods.Items) > 0 {
+		for _, pod := range pods.Items {
+			return pod.Namespace, nil
+		}
+	}
+	return metav1.NamespaceAll, fmt.Errorf("unable to find the synopsys operator namespace due to %+v", err)
+}
+
+// IsDeleteOperatorNamespace returns whether the operator namespace can be delete or not
+func IsDeleteOperatorNamespace(clientset *kubernetes.Clientset, namespace string) bool {
```

Comment:
This is really just a check whether the operator is in Cluster scope.  if in cluster scope, it is ok to delete, otherwise no.

Reply:
This method will be called during destroy operation. If clusterscope or namescope, it will check whether anything is running in the namespace. Even for clusterscope, the user can create a Black Duck instance same as Operator. So validating for both the cases should be good.

Reply:
We never want to delete a ns, or try to, in ns scope.  It's debatable whether we want to worry about a user putting a black duck instance in the same ns as the operator.  To do that they would have to launch the bd instance w/o using the cluster scoped operator.

Reply:
done